vignettes/Treatment.Rmd

```diff
@@ -15,50 +15,126 @@ knitr::opts_chunk$set(
+
+Each drug parameter set is a vector of length four, that represent the drug efficacy, the infectiousness following treatment relative to an untreated infection, and parameters that determine the protection against reinfection ($P$): shape ($w$) and scale ($\lambda$). The decay of protection against infection follows a weibull distribution as follows:
+
+$$P = e^{{(-t/\lambda)}^w}$$
+
+The $P$ for each drug following treatment through time is plotted below.
+
+```{r, echo = FALSE, fig.align = 'center', fig.height = 5, fig.width = 7}
+cols  <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
+
+calc_P <- function(w, lambda, t){
+  P <- exp(-(t/lambda)^w)
+}
+
+t <- 1:70
+
+P_matrix <- lapply(list(AL_params, DHA_PQP_params, SP_AQ_params), function(x){
+  calc_P(w = x[3], lambda = x[4], t = t)
+})
+
+plot(x = t, y = P_matrix[[1]], type = "l", col = cols[1],
+     xlab = "Days", ylab = expression(paste("Protection against reinfection (",italic(P),")")))
+grid()
+
+# 
+invisible(sapply(2:3, function(x){
+  points(x = t, y = P_matrix[[x]], type = "l", col = cols[x])
+}))
+
+# Add legend
+legend("topright", legend = c("AL","DHA-PQP","SP-AQ"), col = cols, lty = 1)
+
+```
+
+For more details, please see:
+
+Okell, L., Cairns, M., Griffin, J. *et al.* Contrasting benefits of different artemisinin combination therapies as first-line malaria treatments using model-based cost-effectiveness analysis. *Nat Commun* **5**, 5606 (2014). <https://doi.org/10.1038/ncomms6606>.
+
+## Functions
+
+Drug parameters can be incorporated into a complete parameter set using the `set_drugs` function which takes the full parameter set and a list of drug parameter sets.
+
+A treatment regimen for each drug can then be described using the `set_clinical_treatment()` function which takes the drug index, a vector of timesteps at which a change in coverage occurs (where the initial coverage is 0 until the first timestep specified) and a vector of coverages for the drug that correspond with the timestep changes, as well as the complete parameter set.
+
+Multiple drugs can be modelled simultaneously, with treatment coverage that can be specified for each drug. This function must be used for each drug included (e.g., if there are two drugs, `set_clinical_treatment()` must be called twice to specify the treatment plan).
+
+
+# Example
+
+We will run a simulation for one year using AL and DHA-PQP treatment regimens. We begin our AL treatment on day 100 at 40% coverage and provide this treatment for 200 days before the treatment stops. DHA-PQP treatment also has 40% coverage, but begins on day 200 and lasts for 100 days, also ending on day 300. Note that the sum of treatment coverages cannot exceed 100%.
+
+Prior to the simulation, the function `set_equilibrium` must also be used to generate equilibrium values for the human and mosquito populations.
```

Comment:
set_equilibrium _may_ also be used to start our simulation run at an equilibrium state